src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -396,6 +402,47 @@ private static void ProcessResult(IAsyncResult asyncResult, AsyncCallback asyncC
         }
 
         activity.Stop();
+
+        TagList tags = default;
+
+        // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md
+        if (emitOldAttributes)
+        {
+            foreach (var tag in activity.TagObjects)
```

Comment:
This adds the overhead of looping through the tags even if someone is not using metrics. Could we add a check here before looping?